batch2/batch/utils.py

```diff
@@ -17,6 +19,53 @@ def parse_cpu_in_mcpu(cpu_string):
     return None
 
 
+conv_factor = {
+    'K': 1000, 'Ki': 1024,
+    'M': 1000**2, 'Mi': 1024**2,
+    'G': 1000**3, 'Gi': 1024**3,
+    'T': 1000**4, 'Ti': 1024**4,
+    'P': 1000**5, 'Pi': 1024**5
+}
+
+
+def parse_memory_in_bytes(memory_string):
+    match = MEMORY_REGEX.fullmatch(memory_string)
+    if match:
+        number = float(match.group(1))
+        suffix = match.group(2)
+        return int(number * conv_factor[suffix])
+    return None
+
+
+def worker_memory_per_core_gb(worker_type):
+    if worker_type == 'standard':
+        m = 3.75
+    elif worker_type == 'highmem':
+        m = 6.5
+    else:
+        assert worker_type == 'highcpu', worker_type
+        m = 0.9
+    return m
+
+
+def worker_memory_per_core_bytes(worker_type):
+    m = worker_memory_per_core_gb(worker_type)
+    return math.ceil(m * 1000**3)  # GCE memory/core are in GB not GiB
+
+
+def memory_bytes_to_cores_mcpu(memory_in_bytes, worker_type):
+    return math.ceil((memory_in_bytes / worker_memory_per_core_bytes(worker_type)) * 1000)
```

Comment:
This was the ceil I asked for.  The thing we want is:

 > cores_mcpu_to_memory_bytes(memory_bytes_to_cores_mcpu(M, T), T) >= M